src/spinegeneric/cli/check_data_consistency.py

```diff
@@ -47,6 +38,62 @@ def read_tsv(path: Path) -> tuple[list[Fieldname], list[Row]]:
         return reader.fieldnames, list(reader)
 
 
+def validate_whitespace(value: str):
+    if value != value.lstrip():
+        raise ValueError("should not have leading whitespace")
+    if value != value.rstrip():
+        raise ValueError("should not have trailing whitespace")
+
+
+def validate_integer(value: str) -> int:
+    validate_whitespace(value)
+    try:
+        return int(value)
+    except ValueError:
+        raise ValueError("should be an integer") from None
+
+
+def validate_sex(value: str):
+    validate_whitespace(value)
+    if value not in ["M", "F", "O"]:
+        raise ValueError("should be one of 'M', 'F', 'O', or 'n/a'")
+
+
+def validate_age(value: str):
+    age = validate_integer(value)
+    if not 18 <= age <= 60:
+        raise ValueError("should be between 18 and 60")
```

Comment:
Is there any merit to generalizing these to be functions that return function objects, such that we can specify configuration arguments (and thus make the `validators` dictionary more self-documented)?

```python
    "sex": validate_list(["M", "F", "O"]),
    "age": validate_range([18, 60]),
```

Kind of like how the old schema functionality was supposed to work?

Reply:
Gotcha! Makes sense. :)